doc/SONiC_long_reset_button_press_hld.md

```diff
@@ -0,0 +1,282 @@
+# Long reset button press HLD
+
+## Table of contents
+
+-   [Revision](#)
+-   [Scope](#scope)
+-   [Definitions/Abbreviations](#definitionsabbreviations)
+-   [Requirements](#requirements-in-this-feature-we-want-to-be-able-to)
+-   [Architecture design](#architecture-design)
+-   [High level design](#high-level-design)
+-   [Performance](#performance)
+-   [Flows](#flows)
+-   [Configuration and management](#configuration-and-management)
+    - [CLI](#cli)
+    - [YANG Model](#yang-model)
+-   [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+-   [Restrictions/Limitations](#restrictionslimitations)
+-   [Testing Requirements/Design](#testing-requirementsdesign)
+    - [Unit Test cases](#unit-test-cases)
+    - [System Test cases](#system-test-cases)
+
+#
+  
+
+# Revision
+
+| **Rev**   | **Date**     |           **Author**            | **Change Description**  |
+|:---------:|:------------:|:-------------------------------:|:------------------------|
+| 1.0       | 04/01/2024   | Azmy Ali (Nvidia SONiC Team)    | Initial public version  |
+
+### 
+
+# Scope
+
+This document describes the design details for supporting long reset button press on a SONiC switch. In this document, we will specify the default behavior
+for SONiC OS on long reset button press and the new service introduced during initializtion flow to handle the functionality applied on long reset button press.
+
+# Definitions/Abbreviations
+
+| **Term**                    | **Description**                                                                      |
+|-----------------------------|--------------------------------------------------------------------------------------|
+| SONiC | Software for Open Networking in the Cloud |
+|   Long reset button press   |   Pressing the physical reset button on the switch not less than 15 seconds          |
+|   short reset button press  |   Pressing the physical reset button on the switch less than 15 seconds              |
+| CLI   | Сommand-line Interface                    |
+| YANG  | Yet Another Next Generation               |
+
+### 
+
+# Requirements Overview
+
+The long reset button press feature implementation shall support the following:
+
+    1. On long reboot press, the OS will restore the local users' configurations, by deleting non-default users and restoring default passwords for default users and expiring  these passwords due to security conecerns.
+    2. Each vendor can decide what is the behavior on long reset button press and add his own implementation.
```

Comment:
What if my requirement is to factory reset the switch upon pressing the long reset button? do we have a flexibility in the code to enhance for the future requirement?

Reply:
I changed the feature name to reset local users' passwords and this triggered by default by long reset button, but you can add the triggering from any condition you see fit. Now you can add a new service that is very similar to this and do your work of reset factory but not for this one..